featuretools/synthesis/dfs.py

```diff
@@ -247,6 +256,33 @@ def dfs(entities=None,
     features = dfs_object.build_features(
         verbose=verbose, return_variable_types=return_variable_types)
 
+    trans_features = []
+    agg_features = []
+    groupby_features = []
+    where_features = []
+
+    base_features = get_base_features(features)
+
+    for feature in base_features:
+        if isinstance(feature, AggregationFeature):
+            if feature.where:
+                where_features.append(feature)
+            else:
+                agg_features.append(feature)
+        elif isinstance(feature, GroupByTransformFeature):
+            groupby_features.append(feature)
+        elif isinstance(feature, TransformFeature):
+            trans_features.append(feature)
+
+    trans_unused = get_unused_primitives(trans_primitives, trans_features)
+    agg_unused = get_unused_primitives(agg_primitives, agg_features)
+    groupby_unused = get_unused_primitives(groupby_trans_primitives, groupby_features)
+    where_unused = get_unused_primitives(where_primitives, where_features)
+
+    unused_primitives = [trans_unused, agg_unused, groupby_unused, where_unused]
+    if any(unused_primitives):
```

Comment:
Should there be a way to turn off the warnings if the user wanted to?

Reply:
> Should there be a way to turn off the warnings if the user wanted to?

That thought crossed my mind at one point too. I don't think we have any other controls in place for users to turn off specific warnings in other places though, but this one does feel a bit more informational than some of the others do.

I don't have a strong opinion one way or the other at this point.

Reply:
Another possibility -- instead of issuing a warning, include this information in the verbose output?

Reply:
I like the idea of including this in the verbose output, based on the nature of the information we are passing to the user.

Reply:
A custom `UnusedPrimitiveWarning` class was created by subclassing `UserWarning`. This will allow users to suppress this warning if desired.